platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java

```diff
@@ -338,6 +350,118 @@ public <ELEMENT> OutputWire<ELEMENT> getSplitOutput() {
+    }
+
+    /**
+     * Get an output wire that will receive a specific type of routed data after being split apart. Data routing is when
+     * a component has multiple outputs, and different things want to receive a subset of that output. Each output is
+     * described by a routing address, which are implemented by enum values.
+     * <p>
+     * This method should only be used for components which have an output type of
+     * {@link com.swirlds.common.wiring.transformers.RoutableData List&lt;RoutableData&gt;}. Calling this method more
+     * than once with different enum classes will throw.
+     *
+     * @param address       an enum value that describes one of the different types of data that can be routed
+     * @param <ROUTER_ENUM> the enum that describes the different types of data handled by this router
+     * @param <DATA_TYPE>   the type of data that travels over the output wire
+     * @return the output wire
+     */
+    @NonNull
+    public <ROUTER_ENUM extends Enum<ROUTER_ENUM>, DATA_TYPE> OutputWire<DATA_TYPE> getSplitAndRoutedOutput(
+            @NonNull final ROUTER_ENUM address) {
+
+        final Class<ROUTER_ENUM> clazz = (Class<ROUTER_ENUM>) address.getClass();
+        return getOrBuildSplitRouter(clazz).getOutput(address);
+    }
+
+    /**
+     * Get the router for this component if one has been built. Build and return a new router if one has not been
+     * built.
+     *
+     * @param routerType    the type of the router
+     * @param <ROUTER_TYPE> the type of the router
+     * @return the router
+     */
+    @NonNull
+    private <ROUTER_TYPE extends Enum<ROUTER_TYPE>> WireRouter<ROUTER_TYPE> getOrBuildRouter(
+            @NonNull final Class<ROUTER_TYPE> routerType) {
+
+        if (splitRouter != null) {
+            throw new IllegalStateException("Only one type of router can be constructed per task scheduler. "
+                    + "This task scheduler already has a router that was built using the split output type, "
+                    + "so a router cannot be created with the unmodified output type.");
+        }
+
+        if (router != null) {
+            if (!router.getRouterType().equals(routerType)) {
+                throw new IllegalArgumentException("Only one type of router can be constructed per task scheduler. "
+                        + "This task scheduler already has a router of type "
+                        + router.getRouterType().getName() + "but an attempt was made to construct a router of type "
+                        + routerType.getName());
+            }
+        } else {
+            router = new WireRouter<>(model, getSchedulerName() + "Router", "data", routerType);
+            getOutputWire().solderTo((InputWire<OUTPUT_TYPE>) router.getInput());
+        }
+
+        return (WireRouter<ROUTER_TYPE>) router;
+    }
+
+    /**
+     * Get the router for this component if one has been built. Build and return a new router if one has not been
```

Comment:
This doc is an exact duplicate of the doc for `getOrBuildRouter`

It should probably address to `split` attribute of this method

Reply:
updated